### PR TITLE
Added support for non-enumerable collections.

### LINKED ISSUE #4030: IsEmpty doesn't work with new .NET6 PriorityQueue
The PriorityQueue class has a `Count` property, but is not `IEnumerable`.

The `EmptyConstraint` should verify that `actual is IEnumerable` instead of assuming it.
If not `IEnumerable` it can check for a `Count` property.

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #4030

Allows for 'collections' with a .Count property, such as the .net 6 PriorityQueue.

### CODE REVIEW

File: src/NUnitFramework/framework/Constraints/CountZeroConstraint.cs
Comment (mikkelbu) [inline src/NUnitFramework/framework/Constraints/CountZeroConstraint.cs:49]: Should we also validate if the type of the property is an `int`. Otherwise I think the cast `ApplyTo` would fail.

**Review by mikkelbu — APPROVED** [review]
Only a single comment. Otherwise it looks good 👍

File: src/NUnitFramework/framework/Constraints/EmptyConstraint.cs
Comment (stevenaw) [inline src/NUnitFramework/framework/Constraints/EmptyConstraint.cs:67]: Thanks for tackling this @manfred-brands . I recently hit this myself with `NameValueCollection` in a netfx project.

If I'm reading the diff right, this should better advise on future edge cases where a non-null yet incompatible type is passed in as it should avoid a cryptic NRE.

Comment (manfred-brands) [reply src/NUnitFramework/framework/Constraints/EmptyConstraint.cs:67]: Yes, previously if always assumed IEnumerble and that resulted in a runtime exception when casting later.

File: src/NUnitFramework/framework/Constraints/CountZeroConstraint.cs
Comment (manfred-brands) [reply src/NUnitFramework/framework/Constraints/CountZeroConstraint.cs]: Added a check on `PropertyType` and added new unit test. Initially I contemplated allowing more types, but I don't think it is worth it.

File: src/NUnitFramework/framework/Constraints/EmptyConstraint.cs
Comment (manfred-brands) [reply src/NUnitFramework/framework/Constraints/EmptyConstraint.cs]: Added test case with `NameValueCollection`.

**Comment by manfred-brands** [comment]
I see two approvals, but the PR is not merged, it anything holding this up?

**Comment by mikkelbu** [comment]
Thanks for the ping @manfred-brands . I have been too busy to keep track of the issues and PRs lately. I'll merge it now. Can you also provide a PR for master?

**Comment by stevenaw** [comment]
Thanks @mikkelbu 
Sorry @manfred-brands I had to step away from my computer before merging after my approval. Thanks for following up 👍

**Pull request merged**
